### DISCUSSION TITLE
nextjs - d3.select(svgRef.current)  issue detect as string.

### DISCUSSION THREAD

**Discussion by chiddekel** [d3/d3#3604] [Q&A] [answered]
On npm run build have error:

      select.js:5
         ? new Selection([[document.querySelector(selector)]], [document.documentElement])
         
detect svgRef.current as a string but is as a type object 
how to fix the d3.select ?

**Comment by Fil** [comment]
Please give some useable context to help reproduce the issue.

**Comment by chiddekel** [comment, marked as answer]
Error with replace useEFfect to useSSE

**Comment by chiddekel** [comment]
> Please give some useable context to help reproduce the issue.

If replace useEffect with useSSE error occured - in another world d3js require hydrated state of components

**Comment by chiddekel** [comment]
For now instead useEffect or useSSE i use srtTimoute
